### ISSUE TITLE
mydumper consumes 100% CPU if disk is full

### ISSUE THREAD

**Issue by rtimmons** [maxbube/mydumper#40]
Heyo - recently stumbled across mydumper and it looks great. Thanks for putting it out there.

Just ran into an issue:

I was running a `mydumper` job and the disk ran out of space. Instead of exiting or emiting error messages, the process hung around at 100% CPU forever (the `/proc/*/stack` file indicated it was waiting on `futex_wait_queue_me`, perhaps all the threads are spinning or something).

For posterity here's the command being run:

```sh
./mydumper/mydumper --version
mydumper 0.9.2, built against MySQL 5.5.54

./mydump/mydumper \
    --triggers --events --routines \
    --no-locks \
    --long-query-guard 31536000 \
    --compress  --compress-protocol \
    --host "$DB_HOST" \
    --user "$DB_USER" \
    --password "$DB_PASS" \
    --rows 500000 \
    --build-empty-files \
    --threads 32 \
    --outputdir "$OUTPUT_DIR" \
    --verbose 3 \
    > "$OUTPUT_DIR".output.log
```

This was compiled from source commit `f79c3e2a` using Amazon Linux AMI release 2014.09.

Not a huge issue but it did take some time to figure out that the disk was full and that was probably the cause.

Hope this helps.

**Comment by maxbube** [comment]
Hi @rtimmons! glad to see more people adopting mydumper, about this issue I don't think there is much I can do if IO is not throwing an error, also is the same behavior as mysqld. And is some cases this could even better is it give you time to release disk space for the backup to complete :)

I will look into this though but with low priority.

**Comment by rtimmons** [comment]
Thanks for response. Infinite CPU is a nice way to buy time I guess 😸. We were able to work around it so no lingering issue from our end - free to resolve if you wanna. If nothing else hopefully this issue will be found by others experiencing a similar issue.

**Comment by chaigang0** [comment]
I also encountered this problem. Unfortunately, mydumper did not have a log indicating that the disk was full or the storage was not enough. It was not until I saw this issue that I realized that the disk may be full.
Hope mydumper can output rich logs to the console.

**Comment by davidducos** [comment]
@chaigang0, I had created https://github.com/mydumper/mydumper/issues/499 for the logs on full disk issues.
